app/Http/Controllers/PlaceController.php

```diff
@@ -45,25 +39,41 @@ public function index($id)
     }
 
     public function getRecommendPlace($id)
-    {       
+    {   
+        # place_keyword tableから$idとplace_idが等しいデータを抽出
         $place_keywords = DB::table('place_keywords')->where('place_id', $id)->get();
```

Comment:
please make use the models to get the keywords

Reply:
I think it is better to use the model when I select the data from the database but it doesn't work well if I wrote it in the model.

Reply:
on the above code you can make use of the `place and keywords` relationship. `getRecommendedPlace($id)` should be `getRecommendedPlace(Place $place)` and from there, you can now get the `keywords` like `$place->keywords`.

Reply:
you can store the `$place->keywords` in to a variable like `$placeKeywords` and your `$keyword_id_list` should be like `$keyword_id_list = $placeKeywords->pluck('id')->unique()->toArray()` and there's no need for the entire 
```
foreach($place_keywords as $place_keyword){
            $keyword_id_list[] = $place_keyword->keyword_id;
}
```

Reply:
and for the `$place_id_list` below should be `$place_id_list =  $placeKeywords->pluck('place_id')->unique()->toArray()`.

and there's no need for 
```
foreach($all_target_data as $target_data){
                $place_id_list[] = $target_data->place_id;
            }
            $target_place_id = array_unique($place_id_list);
```